File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans=0;
    private void helper(TreeNode node,int sum){
        if(node==null)
            return;
        if(node.left==null && node.right==null)
            ans+=sum*10+node.val;
        sum = sum*10+node.val;
        helper(node.left,sum);
        helper(node.right,sum);
    }
    public int sumNumbers(TreeNode root) {
        helper(root,0);
        return ans;
    }
}

File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    void sumNumbersHelper(TreeNode* node,int sum){
        if(node==nullptr)
            return;
        if(node->left ==nullptr && node->right==nullptr)
            ans+=sum*10+node->val;
        sum = sum*10+node->val;
        sumNumbersHelper(node->left,sum);
        sumNumbersHelper(node->right,sum);
    }
public:
    int ans=0;
    int sumNumbers(TreeNode* root) {
        sumNumbersHelper(root,0);
        return ans;
    }
};